packages/components/src/range-control/styles/range-control-styles.ts

```diff
@@ -290,13 +306,27 @@ export const Tooltip = styled.span< TooltipProps >`
 	) }
 `;
 
+const inputNumberWidth = ( {
+	size = 'default',
+}: Pick< NumberControlProps, 'size' > ) => {
+	const sizes = {
+		small: space( 16 ),
+		default: space( 16 ),
+		'__unstable-large': space( 20 ),
+	};
+
+	return css`
+		width: ${ sizes[ size ?? 'default' ] };
+	`;
+};
+
 // @todo: Refactor RangeControl with latest HStack configuration
 // @wordpress/components/ui/hstack.
 export const InputNumber = styled( NumberControl )`
 	display: inline-block;
 	font-size: 13px;
 	margin-top: 0;
-	width: ${ space( 16 ) } !important;
+	${ inputNumberWidth };
```

Comment:
Would it be better/possible to use the `__unstableInputWidth` prop on NumberControl rather than a CSS override?

Reply:
That seems to work well for me when testing in Storybook. Was 6fec45f139 along the lines you were thinking? Just to make sure I've implemented it properly.

If so, it has the added benefit of simplifying all of that `inputWidth` logic and typing!